### PR TITLE
API test for getUser

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
API test for getUser
This PR has some skipped scenarios which will pass only after resolving issue #31276

## Related Issue
#31533 

## How Has This Been Tested?
Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Acceptance test

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/acceptance/features/apiProvisioning-v1/getUser.feature [hunk 1]
```diff
@@ -15,4 +15,36 @@ So that I can get information about user
...
+		And the HTTP status code should be "401"
+
+	Scenario: normal user tries to get the user information
+		Given user "newuser" has been created
+		And user "anotheruser" has been created
+		When user "anotheruser" sends HTTP method "GET" to API endpoint "/cloud/users/newuser"
+		Then the OCS status code should be "997"
+		And the HTTP status code should be "401"
+
+		Scenario: normal user tries to get his information
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v1/getUser.feature]: indent problem

File: tests/acceptance/features/apiProvisioning-v1/getUser.feature [hunk 2]
```diff
@@ -15,4 +15,36 @@ So that I can get information about user
...
+		Given user "subadmin" has been created
+		And user "newuser" has been created
+		And group "newgroup" has been created
+		And user "newuser" has been added to group "newgroup"
+		And user "subadmin" has been made a subadmin of group "newgroup"
+		When user "subadmin" sends HTTP method "GET" to API endpoint "/cloud/users/newuser"
+		Then the OCS status code should be "100"
+		And the HTTP status code should be "200"
+
+	Scenario: subadmin gets information of the user not in his group
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v1/getUser.feature]: s/gets/tries to get/
s/the user/a user/

File: tests/acceptance/features/apiProvisioning-v1/getUser.feature [hunk 3]
```diff
@@ -15,4 +15,36 @@ So that I can get information about user
 	Scenario: Getting a not existing user
 		When user "admin" sends HTTP method "GET" to API endpoint "/cloud/users/test"
 		Then the OCS status code should be "998"
+		And the HTTP status code should be "200"
+
+	Scenario: subadmin gets information of the user in his group
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v1/getUser.feature]: s/the user/a user/

File: tests/acceptance/features/apiProvisioning-v1/getUser.feature [hunk 4]
```diff
@@ -15,4 +15,36 @@ So that I can get information about user
...
+	Scenario: subadmin gets information of the user not in his group
+		Given user "subadmin" has been created
+		And user "newuser" has been created
+		And group "newgroup" has been created
+		And user "subadmin" has been made a subadmin of group "newgroup"
+		When user "subadmin" sends HTTP method "GET" to API endpoint "/cloud/users/newuser"
+		Then the OCS status code should be "997"
+		And the HTTP status code should be "401"
+
+	Scenario: normal user tries to get the user information
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v1/getUser.feature]: normal user tries to get information of another user

File: tests/acceptance/features/apiProvisioning-v1/getUser.feature (hunk 1 above)
Comment (phil-davis) [reply tests/acceptance/features/apiProvisioning-v1/getUser.feature]: normal user gets his own information

**Review by phil-davis — COMMENTED** [review]
Just words in the scenario descriptions. Make the same changes in the v1 and v2 feature files.

We can make a standard - if the request is expected to be successful then we say:
"user gets..."

If the request is expected to not be possible, then we say"
"user tries to get..."

Then, when reading the scenario description, people immediately understand what to expect.

**Review by phil-davis — APPROVED** [review]

**Comment by paurakhsharma** [comment]
Backport on #31387

**Pull request merged**